### PR TITLE
Add the new OpenTK FAQ

### PR DESCRIPTION
**Pull request by seanofw**
This pull request contains the the long-awaited new OpenTK FAQ!

I reread it again today, and I didn't find any major defects or issues.  So let's finally get this onto the website, and hopefully it can make a difference for us in supporting OpenTK :)

### CODE REVIEW

**Review by frederikja163 — APPROVED** [review]
Im not gonna read through this again, i already read the many drafts before this and i trust you haven't changed anything since the latest draft. So its good to go for me! I will leave it here untill i get home next week and have my pc for a build.

File: faq.md
Comment (NogginBops) [inline faq.md:35]: OpenTK 4 only runs on .net core 3.1+ iirc.

File: faq.md
Comment (NogginBops) [inline faq.md:77]: `to reduce overhead even further.` is misleading as it will have the same overhead as the optimized IL (and maybe even slightly worse as we are not optimizing at IL anymore, but we will try to avoid that).

File: faq.md
Comment (NogginBops) [inline faq.md:123]: I think we want to have a clarification here that `OpenGL4` doesn't mean OpenGL version 4 but instead just means that it only exposes the modern functions that you should use, and explaining that it's an unfortunate naming scheme that will be left as is for OpenTK 3 & 4, but will be fixed in OpenTK 5.

File: faq.md
Comment (NogginBops) [inline faq.md:204]: I think we want to mention that `All` contains all of the enums, so if you don't find the enum you are looking for in a group you should open an issue and use the `All` enum as a workaround.

File: faq.md
Comment (NogginBops) [inline faq.md:208]: I also think that we should explicitly say that our tutorials are based on https://learnopengl.com

File: faq.md
Comment (NogginBops) [inline faq.md:287]: This section is taking up quite a big part of the FAQ and it's pretty in the weeds about implementation, I feel like it might be better to just make this it's own page and link to it here.
Then we can either take Vassals gist and maintain it ourselves or just link to it here.

File: faq.md
Comment (NogginBops) [inline faq.md:582]: I think we need practical guide here too about how you get consistent multiplication order etc, but I can write that at a later time. It doesn't need to be part of this PR.

File: faq.md
Comment (NogginBops) [inline faq.md:588]: `NativeWindow` is not the lowest-level thing in OpenTK 4 (`GLFW.` is), and the split between `NativeWindow` and `GameWindow` isn't as clear as we would like it to be, but this explanation is probably ok.

**Review by NogginBops — CHANGES_REQUESTED** [review]
Very nice FAQ! 
There are a few small things that I think needs to be changed before merging.
Fantastic works, incredibly appreciated!

File: faq.md
Comment (seanofw) [reply faq.md:35]: I don't know that for a fact, but I've removed .NET Core 2.x from the list anyway; anyone using .NET Core is likely on a newer version of it anyway and isn't likely going to be ask why .NET Core 2.x isn't supported.

File: faq.md
Comment (seanofw) [reply faq.md:77]: I replaced it with `use the new C# function pointers to accomplish the same thing.`  I think that reads better.

File: faq.md
Comment (seanofw) [reply faq.md:123]: I changed it to this:

> Which ones to use depends on which flavors of OpenGL/CL/AL you’re targeting.  We generally recommend with OpenGL that you should use `OpenGL4` unless you need the legacy fixed-function pipeline (FFP); the `OpenGL4` namespace works for all OpenGL versions, but it excludes the FFP functions and their overhead.  (In OpenGL 5, we plan to rename these namespaces to make this distinction more obvious.)

Comment (NogginBops) [reply faq.md:123]: Good enough for me 🙂

File: faq.md
Comment (seanofw) [reply faq.md:204]: This took a bit of a rewrite; I hadn't looked closely at what `All` was originally, and erroneously assumed it was something it wasn't.  Here's the revised text:

> - The constants are grouped into strong `enum` types like `ErrorCode` and `ShaderType` — they’re not just a big pile of unrelated values like they are in C.
> - The names are changed from `SHOUT_CASE` to `PascalCase` to match C# style conventions.
> - The constants' `GL_` prefixes have been removed for readability.
>
> There is also a special enum type, `All`, that contains *all* of the GL constants; this can simplify porting code from other languages, and it ensures that all of the OpenGL constants are available.  `All` should be avoided in favor of more specialized enum types where possible.

File: faq.md
Comment (seanofw) [reply faq.md:208]: Easy fix:

> You can learn more about OpenGL in depth over at [Learn OpenGL](https://learnopengl.com/); our tutorials are based on theirs.

File: faq.md
Comment (seanofw) [reply faq.md:287]: It's in the weeds, true, but error-handling is such an important topic for *any* library that I really think we should keep it.  OpenGL tutorials often gloss over error-handling; we should not, and should not just teach people how to use OpenGL, but how to use it *well*.  "How do I handle errors?" is a reasonable question that's part of "How do I use OpenGL well?" — and a good answer to it takes space.  We can afford a few extra bytes and lines of text to include a good answer.

File: faq.md
Comment (seanofw) [reply faq.md:582]: Agreed.  We do need a good guide on that, describing all the different ways different things handle matrixes, but that's outside the scope of this.  If you write it up, we can add a link to it.

File: faq.md
Comment (NogginBops) [reply faq.md:204]: I think we should have something that mentions how you can cast an `All` enum to the type of enum you need. As that is apparently something that a lot of people don't really know that you can do, and it would be especially nice if we mentioned that.
For example see the workaround in https://github.com/opentk/opentk/issues/688
`GL.GetFloat((GetPName)All.MaxTextureMaxAnisotropy);`

These are actually relevant to a few well used enums, specifically anisotropic filtering texture parameter and some `SizedInternalFormats` that are missing (see https://github.com/opentk/opentk/issues/752)

File: faq.md
Comment (seanofw) [reply faq.md:588]: The split isn't always clear, but that's why it's followed up with `you probably want to use`.

I've added this addendum to the first bullet point:

> `NativeWindow` is the lowest-level class.  It includes the bare minimum of required mechanics to open and display a window, and not much else.  It’s small, light, and fast, and good for situations where you don’t need an update/render loop or you intend to write most of the update/render logic yourself.  (Note that `NativeWindow` is not the lowest *level*, merely the lowest level in managed .NET:  `NativeWindow` wraps native functionality provided by GLFW.)

**Comment by seanofw** [comment]
@NogginBops I think I've updated everything that you noted that needed to be updated.  Let me know what you think of the result.

File: faq.md
Comment (NogginBops) [reply faq.md:289]: I agree that we should have an answer, but I think that answer can be a link to another page. Though idk if the website project would handle that gracefully.
We can leave it like this for this PR and see if we want to change it later.

File: faq.md
Comment (seanofw) [reply faq.md]: I added a short parenthetical about casting `All`:

> There is also a special enum type, `All`, that contains *all* of the GL constants; this can simplify porting code from other languages, and it ensures that all of the OpenGL constants are available.  `All` should be avoided in favor of more specialized enum types where possible (but it isn't *always* possible, so don't forget you can cast from `All` to `int` or to the enum type you need!).

**Comment by seanofw** [comment]
@NogginBops All right, I think I've covered everything you suggested:  Let me know if there's anything else you think needs to be changed.

**Review by NogginBops — APPROVED** [review]
I'm going to leave this to @frederikja163 to merge as he has more knowledge of how the website works and what needs to be done to make this page to live.

**Pull request merged**
